### PR TITLE
Implement a function to compute the nth Catalan number

### PR DESCRIPTION
**Pull request by UTSAVS26**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/test/java/com/thealgorithms/maths/CatalanNumbersTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test class for CatalanNumbers
+ */
+class CatalanNumbersTest {
+
+    @Test
+    void testCatalanNumbers() {
```
Comment (saahil-mahato) [inline src/test/java/com/thealgorithms/maths/CatalanNumbersTest.java:15]: Better to convert this into a parameterized test,

**Review by saahil-mahato — CHANGES_REQUESTED** [review]

**Comment by UTSAVS26** [comment]
@saahil-mahato changes made as per requested, thank you for your time.

**Comment by UTSAVS26** [comment]
@alxkm please review my PR.

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**